src/CommunityToolkit.Mvvm.SourceGenerators/Extensions/CompilationExtensions.cs

```diff
@@ -87,4 +91,37 @@ public static bool HasAccessibleTypeWithMetadataName(this Compilation compilatio
 
         return false;
     }
+
+    /// <summary>
+    /// Tries to build a map of <see cref="INamedTypeSymbol"/> instances form the input mapping of names.
+    /// </summary>
+    /// <typeparam name="T">The type of keys for each symbol.</typeparam>
+    /// <param name="compilation">The <see cref="Compilation"/> to consider for analysis.</param>
+    /// <param name="typeNames">The input mapping of <typeparamref name="T"/> keys to fully qualified type names.</param>
+    /// <param name="typeSymbols">The resulting mapping of <typeparamref name="T"/> keys to resolved <see cref="INamedTypeSymbol"/> instances.</param>
+    /// <returns>Whether all requested <see cref="INamedTypeSymbol"/> instances could be resolved.</returns>
+    public static bool TryBuildNamedTypeSymbolMap<T>(
+        this Compilation compilation,
+        IEnumerable<KeyValuePair<T, string>> typeNames,
```

Comment:
- Curious why this is not a dictionary?
- Assuming all analyzers need *all* types in `typeNames`, LGTM.

Reply:
> "Curious why this is not a dictionary?"

No reason, it was just to give callers more flexibility. I could also make this a dictionary, yeah.

> "Assuming all analyzers need all types in typeNames, LGTM."

Yeah, this would be used by each analyzer to only query all potentially used symbols once at compilation start.